sklearn/utils/estimator_checks.py

```diff
@@ -333,42 +334,89 @@ def _construct_instance(Estimator):
     return estimator
 
 
+def _get_estimator_name(estimator):
+    """Get name of estimator instance or class"""
+    if isinstance(estimator, type):
+        # got a class
+        return estimator.__name__
+    # got an instance
+    return type(estimator).__name__
```

Comment:
Could you re-use this function in `_set_check_estimator_ids` where there is currently redundant code?

Reply:
`_set_check_estimator_ids` gets the ID from an estimator using `str`.

I removed this function to reduce the size of the diff, since it is not really needed anymore.